### PR TITLE
[CodeGen][NFC] Use backend defined configuration pipeline in tests.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Comment by hanhanW** [comment]
I think it is better to scope the tests using backend related passes and pipelines.

File: compiler/src/iree/compiler/Codegen/LLVMGPU/test/gpu_set_num_workgroups.mlir [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-codegen-materialize-user-configs, iree-llvmgpu-select-lowering-strategy)))" \
+// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-codegen-llvmgpu-configuration-pipeline, iree-llvmgpu-select-lowering-strategy)))" \
```
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/test/gpu_set_num_workgroups.mlir]: `iree-codegen-llvmgpu-configuration-pipeline` should run `iree-llvmgpu-select-lowering-strategy` as the last pass, so you can just drop the selection pass (same for all the other instances).

**Review by qedawkins — CHANGES_REQUESTED** [review]
This looks like a good change! Just one comment

File: compiler/src/iree/compiler/Codegen/LLVMGPU/test/gpu_set_num_workgroups.mlir (hunk 1 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/test/gpu_set_num_workgroups.mlir]: Ah, good point. I was not aware of it, thanks!

**Review by qedawkins — APPROVED** [review]
Nice

**Pull request merged**